src/batch-client/index.ts

```diff
@@ -551,6 +551,42 @@ export class ZimbraBatchClient {
 			}
 		}).then(res => normalize(Folder)(res.folder[0].folder));
 
+	public uploadMessage = (message: string) => {
+		const contentDisposition = 'attachment';
+		const filename = 'message.eml';
+		const contentType = 'message/rfc822';
+
+		return fetch(`${this.origin}/service/upload?fmt=raw`, {
+			method: 'POST',
+			body: message,
+			headers: {
+				'Content-Disposition': `${contentDisposition}; filename="${filename}"`,
+				'Content-Type': contentType
+			},
+			credentials: 'include'
```

Comment:
`credentials: 'same-origin'` is a little safer, but probably does not make a difference.